src/main/java/org/dswarm/graph/Neo4jProcessor.java

```diff
@@ -298,25 +376,11 @@ public void addLabel(final Node node, final String labelString) {
 		}
 	}
 
-	public Relationship getStatement(final String hash) throws DMPGraphException {
-
-		IndexHits<Relationship> hits = statementHashes.get(GraphStatics.HASH, hash);
-
-		if (hits != null && hits.hasNext()) {
-
-			final Relationship rel = hits.next();
-
-			hits.close();
-
-			return rel;
-		}
+	public boolean checkStatementExists(final long hash) throws DMPGraphException {
 
-		if (hits != null) {
+		return !(statementHashes == null && tempStatementHashes == null) && (tempStatementHashes != null && tempStatementHashes.contains(hash)
+				|| statementHashes != null && statementHashes.contains(hash));
```

Comment:
I'd probably better to perform the null checks once in the constructor. The field is final, so we can safely assume a non-null value here and safe us some branching.